src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -106,18 +106,14 @@ private void ClearCache()
             {
                 callingType = callingType.GetGenericTypeDefinition();
             }
+
             if (name == SpecialNames.Indexer)
             {
-                // What about named indexers?
-                if (callingType == typeof(string))
-                {
-                    name = "Chars";
-                }
-                else
-                {
-                    name = "Item";
-                }
+                // If we don't find an indexer name for this type, use SpecialNames.Indexer as a key on the
+                // empty results we'll get, so that that gets cached.
```

Comment:
Subordinating *that* followed by demonstrative *that* rather than a duplicate, but I'll rephrase.